include/fmt/core.h

```diff
@@ -545,6 +545,11 @@ struct typed_value : value<Context> {
 template <typename Context, typename T>
 FMT_CONSTEXPR basic_format_arg<Context> make_arg(const T &value);
 
+#if FMT_GCC_VERSION >= 406
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wuseless-cast"
+#endif
```

Comment:
I suggest either fixing the warning or not setting `-Wuseless-cast` in CMake config if the warning is too noisy. Similarly in other cases since it's better not to pollute the code with numerous ifdefs for spurious warnings.

Reply:
Not enabling the warning during testing is only ignoring the problem. People may and will have that warning enabled when including fmt, and they'll have to resort to similar pragma-trickery in order to silence the compiler about issues they have no control over. It's better to explicitly disable a single warning for a small piece of code.

For that specific case, though, fixing the warning should be possible with ease. I don't think the same can be said for the other instances.